Sources/TuistGenerator/Generator/SchemesGenerator.swift

```diff
@@ -269,6 +278,7 @@ final class SchemesGenerator: SchemesGenerating {
 
         let onlyGenerateCoverageForSpecifiedTargets = codeCoverageTargets.count > 0 ? true : nil
 
+        let disableMainThreadChecker = !testAction.diagnosticsOptions.contains(.mainThreadChecker)
```

Comment:
I thought about using the affermative (enable) instead of negative (disable) `disableMainThreadChecker` -> `isMainThreadCheckerEnabled` since I think that's more readable. But I see that's the name used in the `XCScheme`

Reply:
`disableMainThreadChecker` is how Xcode calls the attribute 😛 . That's why I took the opportunity to make it positive from our API `. mainThreadChecker`